dhis-2/dhis-api/src/main/java/org/hisp/dhis/dashboard/Dashboard.java

```diff
@@ -67,11 +67,35 @@ public class Dashboard extends BaseNameableObject implements MetadataObject {
   /** Allowed filter dimensions (if any) which may be used for the dashboard. */
   private List<String> allowedFilters = new ArrayList<>();
 
+  /**
+   * Provider of embedded dashboards, is never <code>null</code>. The default value is {@link
+   * EmbeddedProvider#NONE} which refers to standard DHIS 2 dashboards. Other values indicate that
+   * this dashboard is sourced from an external provider and embedded in DHIS 2.
+   */
+  private EmbeddedProvider embeddedProvider;
```

Comment:
We could have an embedded class, like `Embedded`, that encapsulated the following:

```
private String id;
private Provider provider;
private Options options;
```

or something similar, if possible. So, we can groups all embedded thing into a parent single object.

Reply:
Also, could we have many embedded providers? If so, we could have a Set<Embedded> of embedded(s).

Reply:
Yes good point, was thinking of how to use an embedded class (meaning @Embedded Hibernate annotation/type), I am just unsure how that will work with our metadata importer, as I don't think we use embedded classes anywhere else.

We could use a class and persist as JSONB. That way we can encapsulate it, though that makes it harder to upgrade with Flyway later.

Reply:
There can only be one embedded "item". If you want many embedded dashboards, you create more DHIS 2 dashboards.

Reply:
I have modified it to use an `EmbeddedDashboard` class with JSONB storage now @maikelarabori .